bundle/config/mutator/override_compute.go

```diff
@@ -38,24 +38,33 @@ func overrideJobCompute(j *resources.Job, compute string) {
 }
 
 func (m *overrideCompute) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
-	if b.Config.Bundle.Mode != config.Development {
+	var diags diag.Diagnostics
+
+	if b.Config.Bundle.Mode == config.Production {
 		if b.Config.Bundle.ClusterId != "" {
-			return diag.Errorf("cannot override compute for an target that does not use 'mode: development'")
+			// Overriding compute via a command-line flag for production works, but is not recommended.
+			diags = diags.Extend(diag.Warningf("overriding compute for a target that uses 'mode: production' is not recommended"))
```

Comment:
Can we extend the warnings to include the modalities through which the compute can be overridden? For example, via `DATABRICKS_CLUSTER_ID`, `--cluster-id`, or by configuring `bundle.cluster_id`.

In some contexts, like when using the Databricks VSCode extension, the cluster override can be accidentally configured, and this helps the user know what "overriding compute" means.

Reply:
+1 to that. Allowing the override through the env var should remain limited to dev mode.